Formula/hut.rb

```diff
@@ -0,0 +1,21 @@
+class Hut < Formula
+  desc "CLI tool for sr.ht"
+  homepage "https://sr.ht/~emersion/hut"
+  url "https://git.sr.ht/~emersion/hut/archive/v0.1.0.tar.gz"
+  sha256 "5af8f1111f9ec1da9a818978eb1f013dfd50ad4311c79d95b0e62ad428ac1c59"
+  license "AGPL-3.0-or-later"
+
+  depends_on "coreutils" => :build # GNU install
+  depends_on "go" => :build
+  depends_on "scdoc" => :build
+
+  def install
+    system "make"
+    system "make", "install", "PREFIX=#{prefix}", "INSTALL=ginstall"
+  end
+
+  test do
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
I updated the test. There is not much I can do without valid credentials, so I simply asserted that missing credentials shows an error message.

I tried using invalid credentials (as opposed to missing credentials), but it looks like there is a bug in `hut` that results in a seg fault in this case. When I have a chance I might look into this and send a patch upstream.